projects/packages/publicize/src/class-publicize-base.php

```diff
@@ -1799,7 +1799,7 @@ public function has_instagram_connection_feature() {
 	 * @return bool
 	 */
 	public function has_mastodon_connection_feature() {
-		return Current_Plan::supports( 'social-mastodon-connection' );
+		return Current_Plan::supports( 'social-mastodon-connection', true );
```

Comment:
As discussed, I think it's fine to leave this as is. The plan data will be refreshed at least every 24 hours on the `jetpack_heartbeat`, when the plan changes, or if the dashboard is visited.

It means the feature won't be available straightaway for everyone but within 24 hours it will be.

Reply:
What do you think, @kraftbj? We'll need to take that delay into consideration when setting up test sites.

Reply:
You can force the update on a test site by going to the Jetpack dashboard or running `do_action( 'jetpack_heartbeat' )` in the wp-cli.

As the check is run on editor load it's probably best to deal with the caching rather than making lots of unnecessary API calls. We've also previously caused some performance issues on WPCOM/Atomic doing that.

Reply:
> You can force the update on a test site by going to the Jetpack dashboard 
Just by visiting the page in WPadmin?

Reply:
What if instead of always refreshing the cache we could refresh it once when actually switching the feature on and off in the backend?

Reply:
> Just by visiting the page in WPadmin?

Yes, that's right. It triggers the heartbeat action I think, but definitely refreshes the plan data.

> What if instead of always refreshing the cache we could refresh it once when actually switching the feature on and off in the backend?

We could do that. There is an API endpoint on the site that is pinged as a plan is activated/deactivated, so we could do that for these sites.

I think it is fine to wait. I don't know if the plan data is refreshed after plugin upgrade, but that would be perfect if it is. Otherwise, the features will become available within 24 hours and if someone contacts us asking where they are, we can tell them to visit the Jetpack Dashboard. That's not ideal, but it's better than refreshing the plan data all the time.